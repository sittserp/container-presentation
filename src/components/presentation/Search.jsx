import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ search, onChange }) => (
  <>
    <label htmlFor="search">Search</label>
    <input
      id="search"
      name="search"
      type="text"
      value={search}
      onChange={onChange} />
  </>
);
Search.propTypes = {
  search: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};
export default Search;
