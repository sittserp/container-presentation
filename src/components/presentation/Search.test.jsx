import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders Search', () => {
    const { asFragment } = render(<Search 
      search="dummy string"
      onChange={jest.fn()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
