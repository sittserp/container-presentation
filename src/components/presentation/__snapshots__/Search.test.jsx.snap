// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<DocumentFragment>
  <label
    for="search"
  >
    Search
  </label>
  <input
    id="search"
    name="search"
    placeholder="Search Here"
    type="text"
    value="dummy string"
  />
</DocumentFragment>
`;
